lib/premailer/adapter/nokogiri.rb

```diff
@@ -78,7 +78,11 @@ def to_inline_css
             Premailer::RELATED_ATTRIBUTES[el.name].each do |css_att, html_att|
               if el[html_att].nil? and not merged[css_att].empty?
                 new_html_att = merged[css_att].gsub(/url\(['"](.*)['"]\)/, '\1').gsub(/;$|\s*!important/, '').strip
-                el[html_att] = css_att.end_with?('color') && @options[:rgb_to_hex_attributes] ? ensure_hex(new_html_att, el) : new_html_att
+                if @options[:rgb_to_hex_attributes] && css_att.end_with?('color')
+                  new_html_att = ensure_hex(new_html_att, el)
+                end
+                el[html_att] = new_html_att
+                merged[css_att] = new_html_att.declarations_to_s
```

Comment:
@avit you had asked why we were doing `.to_s` here and not on the other element.
`el[html_att]` gets transformed to string on line 98.

If we don't do this, we are trying to add an object into the html. no bueno.